src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -174,8 +160,20 @@ public async Task<ProvenBlockHeader> GetAsync(int blockHeight)
         {
             using (new StopwatchDisposable(o => this.performanceCounter.AddQueryTime(o)))
             {
+                lock (this.lockObject)
+                {
+                    if (this.PendingBatch.TryGetValue(blockHeight, out ProvenBlockHeader headerFromBatch))
```

Comment:
Ok yes this fixes the other issue I reported that if cache is depleted the header might be gone.
Unfortunately we need to lock on every call but its cheap as its a dictionary read.

Reply:
Could you please close that other issue if it's fixed? 
Wasn't able to find it